components/script/dom/globalscope.rs

```diff
@@ -2128,6 +2131,17 @@ impl GlobalScope {
             .borrow_mut()
             .create_pipeline_layout_id(backend)
     }
+
+    pub fn wgpu_create_shader_module_id(&self, backend: Backend) -> ShaderModuleId {
+        self.gpu_id_hub
+            .borrow_mut()
+            .create_shader_module_id(backend)
+    }
+    pub fn wgpu_create_compute_pipeline_id(&self, backend: Backend) -> ComputePipelineId {
+        self.gpu_id_hub
```

Comment:
A bit surprised you are able to use the `RefCell` here. In Gecko implementation, multiple threads in the same content process may want to create their own resource IDs.

Reply:
@imiklos Please file an issue about this - it sounds like when we create an ID, we could have multiple threads that end up creating the same ID independently. Possible we need to share a single gpu_id_hub between all threads using Arc & Mutex?